# This patch has been submitted upstream but not yet accepted:
# https://bugzilla.gnome.org/show_bug.cgi?id=734017

--- libxml2-2.9.1/python/tests/sync.py~	2013-03-30 07:34:34.000000000 -0700
+++ libxml2-2.9.1/python/tests/sync.py	2014-07-30 14:27:54.057901481 -0700
@@ -51,7 +51,7 @@
 reference = "startDocument:startElement foo None:startElement bar2 None:endElement bar2:"
 if log != reference:
     print("Error got: %s" % log)
-    print("Exprected: %s" % reference)
+    print("Expected: %s" % reference)
     sys.exit(1)
 
 log=""
@@ -63,7 +63,7 @@
 reference = "startDocument:startElement foo None:startElement bar2 None:endElement bar2:"
 if log != reference:
     print("Error got: %s" % log)
-    print("Exprected: %s" % reference)
+    print("Expected: %s" % reference)
     sys.exit(1)
 
 log=""
@@ -75,7 +75,7 @@
 reference = "startDocument:startElement foo None:startElement bar2 None:"
 if log != reference:
     print("Error got: %s" % log)
-    print("Exprected: %s" % reference)
+    print("Expected: %s" % reference)
     sys.exit(1)
 
 log=""
@@ -84,10 +84,11 @@
 ctxt.parseChunk(chunk, len(chunk), 0)
 ctxt=None
 
-reference = "startDocument:startElement foo None:startElement bar2 {'a': '1', 'b': '2'}:endElement bar2:"
-if log != reference:
+reference1 = "startDocument:startElement foo None:startElement bar2 {'a': '1', 'b': '2'}:endElement bar2:"
+reference2 = "startDocument:startElement foo None:startElement bar2 {'b': '2', 'a': '1'}:endElement bar2:"
+if log not in (reference1, reference2):
     print("Error got: %s" % log)
-    print("Exprected: %s" % reference)
+    print("Expected: %s" % reference)
     sys.exit(1)
 
 log=""
@@ -96,10 +97,11 @@
 ctxt.parseChunk(chunk, len(chunk), 0)
 ctxt=None
 
-reference = "startDocument:startElement foo None:startElement bar2 {'a': '1', 'b': '2'}:"
-if log != reference:
+reference1 = "startDocument:startElement foo None:startElement bar2 {'a': '1', 'b': '2'}:"
+reference2 = "startDocument:startElement foo None:startElement bar2 {'b': '2', 'a': '1'}:"
+if log not in (reference1, reference2):
     print("Error got: %s" % log)
-    print("Exprected: %s" % reference)
+    print("Expected: %s" % reference)
     sys.exit(1)
 
 log=""
@@ -108,10 +110,11 @@
 ctxt.parseChunk(chunk, len(chunk), 0)
 ctxt=None
 
-reference = "startDocument:startElement foo None:startElement bar2 {'a': '1', 'b': '2'}:endElement bar2:"
-if log != reference:
+reference1 = "startDocument:startElement foo None:startElement bar2 {'a': '1', 'b': '2'}:endElement bar2:"
+reference2 = "startDocument:startElement foo None:startElement bar2 {'b': '2', 'a': '1'}:endElement bar2:"
+if log not in (reference1, reference2):
     print("Error got: %s" % log)
-    print("Exprected: %s" % reference)
+    print("Expected: %s" % reference)
     sys.exit(1)
 
 log=""
@@ -123,7 +126,7 @@
 reference = "startDocument:startElement foo None:"
 if log != reference:
     print("Error got: %s" % log)
-    print("Exprected: %s" % reference)
+    print("Expected: %s" % reference)
     sys.exit(1)
 
 # Memory debug specific
